docs: bold-updates (#1988)
OffchainLabs/arbitrum-docs: arbitrum-docs/how-arbitrum-works/06-optimistic-rollup.mdx

_Rollup_

> Arbitrum is a Rollup, which means that the inputs to the chain -- the messages that are put into the inbox -- are all recorded on a parent chain (for Arbitrum One, the Ethereum chain) as calldata. For the purposes of this article, we also refer to the parent chain as L1, and the rollup chain as L2.

Reviewer: ```suggestion
Arbitrum is a Rollup, which means that the inputs to the chain -- the messages that are put into the inbox -- are all recorded on a parent chain (for Arbitrum One, the Ethereum chain) as calldata. For the purposes of this article, we also refer to the parent chain as L1, and the Rollup chain as L2.
```

> Arbitrum is a Rollup, which means that the inputs to the chain -- the messages that are put into the inbox -- are all recorded on a parent chain (for Arbitrum One, the Ethereum chain) as calldata. For the purposes of this article, we also refer to the parent chain as L1, and the Rollup chain as L2.

Reviewer 2: ```suggestion
Arbitrum is a Rollup, which means that the inputs to the chain -- the messages that are put into the inbox -- are all recorded on a parent chain (for Arbitrum One, the Ethereum chain) as calldata or blobs (or another format, depending on the configuration of the chain). For the purposes of this article, we also refer to the parent chain as L1, and the Rollup chain as L2.
```

Because of this, everyone has the information they would need to determine the current correct state of the chain -- they have the full history of the inbox, and the results are uniquely determined by the inbox history, so they can reconstruct the state of the chain based only on public information, if needed.

This also allows anyone to be a full participant in the Arbitrum protocol, to run an Arbitrum node or participate as a <a data-quicklook-from="validator">Validator</a>. Nothing about the history or state of the chain is a secret.

_Optimistic_

> Arbitrum is optimistic, which means that Arbitrum advances the state of its chain by letting any party (a “validator”) post on Layer 1 a Rollup block that the party claims is correct, and then giving everyone else a chance to <a data-quicklook-from="challenge">Challenge</a> that claim. If the <a data-quicklook-from="challenge-period">Challenge Period</a> (6.4 days) passes and nobody has challenged the claimed Rollup block, Arbitrum confirms the Rollup block as correct. Arbitrum uses an efficient dispute resolution protocol to identify which party is lying if someone challenges the claim during the challenge period. A brief overview of the protocol is included below; for more details, see the [Bold dispute protocol](/how-arbitrum-works/bold/gentle-introduction.mdx) section of the documentation. After the protocol finishes, any liar or liars will forfeit their deposits, and the truth-teller(s) will take part of that deposit as a reward for their efforts (some of the deposit is burned, guaranteeing that the liar is punished even if there's some collusion going on).

Author: I think I found a few places where caps potentially could be out of place, hopefully I captured them all.

> Arbitrum is optimistic, which means that Arbitrum advances the state of its chain by letting any party (a “validator”) post on Layer 1 a Rollup block that the party claims is correct, and then giving everyone else a chance to <a data-quicklook-from="challenge">Challenge</a> that claim. If the <a data-quicklook-from="challenge-period">Challenge Period</a> (6.4 days) passes and nobody has challenged the claimed Rollup block, Arbitrum confirms the Rollup block as correct. Arbitrum uses an efficient dispute resolution protocol to identify which party is lying if someone challenges the claim during the challenge period. A brief overview of the protocol is included below; for more details, see the [Bold dispute protocol](/how-arbitrum-works/bold/gentle-introduction.mdx) section of the documentation. After the protocol finishes, any liar or liars will forfeit their deposits, and the truth-teller(s) will take part of that deposit as a reward for their efforts (some of the deposit is burned, guaranteeing that the liar is punished even if there's some collusion going on).

Reviewer 2: Lots of corrections here
```suggestion
Arbitrum chains are Optimistic, which means that Arbitrum chains advances the state of its chain by letting any party (a “proposer”) post on its parent chain an assertion that the party claims is correct (alongside a bond), and then giving everyone else a chance to <a data-quicklook-from="challenge">Challenge</a> that claim (with an accompanying bond). If the <a data-quicklook-from="challenge-period">Challenge Period</a> (6.4 days) passes and nobody has challenged the claimed assertion, the protocol confirms the assertion as correct. Arbitrum chains use an efficient dispute resolution protocol to identify which party is lying if the assertion is challenged during the challenge period. A brief overview of the protocol is included below; for more details, see the [Bold dispute protocol](/how-arbitrum-works/bold/gentle-introduction.mdx) section of the documentation. After the protocol finishes, any liar or liars will forfeit their bonds, and the truth-teller(s) will take part of that deposit as a reward for their efforts (most of the bond is taken away by the chain owner, guaranteeing that the liar is punished even if there's some collusion going on).
```

> Arbitrum is optimistic, which means that Arbitrum advances the state of its chain by letting any party (a “validator”) post on Layer 1 a rollup block that that party claims is correct, and then giving everyone else a chance to <a data-quicklook-from="challenge">Challenge</a> that claim. If the <a data-quicklook-from="challenge-period">Challenge Period</a> (6.4 days) passes and nobody has challenged the claimed rollup block, Arbitrum confirms the rollup block as correct. If someone challenges the claim during the challenge period, then Arbitrum uses an efficient dispute resolution protocol to identify which party is lying. A brief overview of the protocol is included below; for more details, see the [Bold dispute protocol](/how-arbitrum-works/bold/gentle-introduction.mdx) section of the documentation. After the protocol finishes, any liar or liars will forfeit their deposits, and the truth-teller(s) will take part of that deposit as a reward for their efforts (some of the deposit is burned, guaranteeing that the liar is punished even if there's some collusion going on).

Reviewer: ```suggestion
Arbitrum is optimistic, which means that Arbitrum advances the state of its chain by letting any party (a “validator”) post on Layer 1 a Rollup block that the party claims is correct, and then giving everyone else a chance to <a data-quicklook-from="challenge">Challenge</a> that claim. If the <a data-quicklook-from="challenge-period">Challenge Period</a> (6.4 days) passes and nobody has challenged the claimed Rollup block, Arbitrum confirms the Rollup block as correct. Arbitrum uses an efficient dispute resolution protocol to identify which party is lying if someone challenges the claim during the challenge period. A brief overview of the protocol is included below; for more details, see the [Bold dispute protocol](/how-arbitrum-works/bold/gentle-introduction.mdx) section of the documentation. After the protocol finishes, any liar or liars will forfeit their deposits, and the truth-teller(s) will take part of that deposit as a reward for their efforts (some of the deposit is burned, guaranteeing that the liar is punished even if there's some collusion going on).
```

There are basically two choices: interactive proving, or re-executing transactions. Arbitrum uses interactive proving, which we believe is more efficient and more flexible. Much of the design of Arbitrum follows from this fact.

::: note

> Zero-knowledge rollups are able to avoid this situation entirely by using ZK proofs to prove directly to the parent chain that the posted result is correct.

Reviewer: ```suggestion
Zero-knowledge Rollups are able to avoid this situation entirely by using ZK proofs to prove directly to the parent chain that the posted result is correct.
```

:::

### Interactive proving

> The idea of interactive proving is that Alice (along with anyone who agrees with her assertion); Bob (along with anyone who agrees with his assertion); and possibly other parties backing other assertions will engage in an "all-vs-all" challenge game. This is a protocol, refereed by an L1 contract, to resolve their dispute with minimal work required from any L1 contract.

Reviewer 2: ```suggestion
The idea of interactive proving is that Alice (along with anyone who agrees with her assertion); Bob (along with anyone who agrees with his assertion); and possibly other parties backing other assertions will engage in an "all-vs-all" challenge game. This is a protocol, refereed by a contract on the parent chain, to resolve their dispute with minimal work required from any contract on the parent chain.
```

> Arbitrum's approach is based on bisection of the dispute. First, Alice posts a claim, covering N steps of execution. If Bob disagrees with her claim, he proceeds by posting his own claim about the result of execution, which we say "rivals" Alice's. To defend her claim, Alice (or someone acting in support of her claim) posts two claims of size N/2 which combine to yield her initial N-step claim. If Bob wants to continue the dispute, he (or an ally acting in support fo his claim) must do the same. Bob (or an ally) will then pick one of Alice's N/2-step claims to challenge. Now the size of the dispute has been cut in half. This process continues, cutting the dispute in half at each stage, until they are disagreeing about a single step of execution. Note that so far the L1 referee hasn't had to think about execution "on the merits". It is only once the dispute is narrowed down to a single step that the L1 referee needs to resolve the dispute by looking at what the instruction actually does and whether Alice's claim about it is correct.

Reviewer: ```suggestion
Arbitrum's approach is based on the bisection of the dispute. First, Alice posts a claim covering `N` steps of execution. If Bob disagrees with her claim, he proceeds by posting his own claim about the result of execution, which we say "rivals" Alice's. To defend her claim, Alice (or someone acting in support of her claim) posts two size `N/2` claims, which combine to yield her initial `N-step` claim. If Bob wants to continue the dispute, he (or an ally supporting his claim) must do the same. Bob (or an ally) will then pick one of Alice's `N/2`-step claims to challenge. Now, the size of the dispute has been cut in half. This process continues, cutting the dispute in half at each stage until they disagree about a single execution step. Note that so far, the L1 referee hasn't had to think about execution "on the merits". Only once the dispute is narrowed down to a single step does the L1 referee need to resolve it by looking at what the instruction does and whether Alice's claim about it is correct.
```

> Arbitrum's approach is based on the bisection of the dispute. First, Alice posts a claim covering `N` steps of execution. If Bob disagrees with her claim, he proceeds by posting his own claim about the result of execution, which we say "rivals" Alice's. To defend her claim, Alice (or someone acting in support of her claim) posts two size `N/2` claims, which combine to yield her initial `N-step` claim. If Bob wants to continue the dispute, he (or an ally supporting his claim) must do the same. Bob (or an ally) will then pick one of Alice's `N/2`-step claims to challenge. Now, the size of the dispute has been cut in half. This process continues, cutting the dispute in half at each stage until they disagree about a single execution step. Note that so far, the L1 referee hasn't had to think about execution "on the merits". Only once the dispute is narrowed down to a single step does the L1 referee need to resolve it by looking at what the instruction does and whether Alice's claim about it is correct.

Reviewer 2: ```suggestion
Arbitrum's approach is based on the bisection of the dispute. First, Alice posts a claim covering `N` steps of execution. If Bob disagrees with her claim, he proceeds by posting his own claim about the result of execution, which we say "rivals" Alice's. To defend her claim, Alice (or someone acting in support of her claim) posts two size `N/2` claims, which combine to yield her initial `N-step` claim. If Bob wants to continue the dispute, he (or an ally supporting his claim) must do the same. Bob (or an ally) will then pick one of Alice's `N/2`-step claims to challenge. Now, the size of the dispute has been cut in half. This process continues, cutting the dispute in half at each stage until they disagree about a single execution step. Note that so far, the parent chain referee hasn't had to think about execution "on the merits". Only once the dispute is narrowed down to a single step does the parent chain referee need to resolve it by looking at what the instruction does and whether Alice's claim about it is correct.
```

:::note

> Again, this is a very high-level overview. Please refer to the [Gentle introduction](/how-arbitrum-works/bold/gentle-introduction.mdx) and [Technical deep dive](/bold/concepts/bold-technical-deep-dive.mdx) under the **Bold dispute protocol** section in the documentation for more details.

Reviewer 2: ```suggestion
Again, this is a very high-level overview. Please refer to the [Gentle introduction](/how-arbitrum-works/bold/gentle-introduction.mdx) and [Technical deep dive](/bold/concepts/bold-technical-deep-dive.mdx) under the **BoLD dispute protocol** section in the documentation for more details.
```

We believe strongly that interactive proving is the superior approach, for the following reasons.

> **More efficient in the optimistic case**: Because interactive proving can resolve disputes larger than one transaction, it can allow a Rollup block to contain only a single claim about the end state of the chain after all of the execution covered by the block. By contrast, re-execution requires posting a state claim for each transaction within the rollup block. With hundreds or thousands of transactions per rollup block, there is a substantial difference in L1 footprint, and the L1 footprint is the main component of the cost of operating an L2.

Reviewer 2: ```suggestion
**More efficient in the optimistic case**: Because interactive proving can resolve disputes larger than one transaction, it can allow an assertion to contain only a single claim about the end state of the chain after all of the execution covered by the block. By contrast, re-execution requires posting a state claim for each transaction within the rollup block. With hundreds or thousands of transactions per rollup block, there is a substantial difference in computational footprint on the parent chain, and the computational element on the parent chain is usually the main component of the cost of operating a rollup.
```

> **More efficient in the optimistic case**: Because interactive proving can resolve disputes that are larger than one transaction, it can allow a rollup block to contain only a single claim about the end state of the chain after all of the execution covered by the block. By contrast, re-execution requires posting a state claim for each transaction within the rollup block. With hundreds or thousands of transactions per rollup block, this is a substantial difference in L1 footprint -- and L1 footprint is the main component of cost of operating an L2.

Reviewer: ```suggestion
**More efficient in the optimistic case**: Because interactive proving can resolve disputes larger than one transaction, it can allow a Rollup block to contain only a single claim about the end state of the chain after all of the execution covered by the block. By contrast, re-execution requires posting a state claim for each transaction within the rollup block. With hundreds or thousands of transactions per rollup block, there is a substantial difference in L1 footprint, and the L1 footprint is the main component of the cost of operating an L2.
```

> **More efficient in the pessimistic case**: In case of a dispute, interactive proving requires the L1 referee contract only to check that Alice and Bob's actions "have the right shape". For example, that Alice has divided her `N`-step claim into two claims half as large. (The referee doesn't need to evaluate the correctness of Alice's claims--Bob does that off-chain.) Only one instruction needs to be re-executed. By contrast, re-execution requires the L1 referee to emulate the execution of an entire transaction.

Reviewer 2: ```suggestion
**More efficient in the pessimistic case**: In case of a dispute, interactive proving only requires the referee contract on the parent chain to check that Alice and Bob's actions "have the right shape". For example, that Alice has divided her `N`-step claim into two claims half as large. (The referee doesn't need to evaluate the correctness of Alice's claims--Bob does that off-chain.) Only one instruction needs to be re-executed. By contrast, re-execution requires the parent chain referee contract to emulate the execution of an entire transaction.
```

> **More efficient in the pessimistic case**: In case of a dispute, interactive proving requires the L1 referee contract only to check that Alice and Bob's actions "have the right shape", for example, that Alice has divided her N-step claim into two claims half as large. (The referee doesn't need to evaluate the correctness of Alice's claims--Bob does that, off-chain.) Only one instruction needs to be re-executed. By contrast, re-execution requires the L1 referee to emulate the execution of an entire transaction.

Reviewer: ```suggestion
**More efficient in the pessimistic case**: In case of a dispute, interactive proving requires the L1 referee contract only to check that Alice and Bob's actions "have the right shape". For example, that Alice has divided her `N`-step claim into two claims half as large. (The referee doesn't need to evaluate the correctness of Alice's claims--Bob does that off-chain.) Only one instruction needs to be re-executed. By contrast, re-execution requires the L1 referee to emulate the execution of an entire transaction.
```

> **Higher per-tx gas limit:** Interactive proving can escape from Ethereum's tight per-transaction gas limit. The gas limit isn't infinite, for obvious reasons, but it can be larger than on Ethereum. As far as Ethereum is concerned, the only downside of a gas-heavy Arbitrum transaction is that it may require an interactive <a data-quicklook-from="fraud-proof">fraud proof</a> with slightly more steps (and only if a fraud proof is actually needed; i.e., in the event of a dispute). By contrast, rerexecution must impose a _lower_ gas limit than Ethereum, because it must be possible to emulate execution of the transaction (which is more expensive than executing it directly) within a single Ethereum transaction.

Reviewer 2: typo
```suggestion
**Higher per-txn gas limit:** Interactive proving can escape from Ethereum's tight per-transaction gas limit. The gas limit isn't infinite, for obvious reasons, but it can be larger than on Ethereum. As far as Ethereum is concerned, the only downside of a gas-heavy Arbitrum transaction is that it may require an interactive <a data-quicklook-from="fraud-proof">fraud proof</a> with slightly more steps (and only if a fraud proof is actually needed; i.e., in the event of a dispute). By contrast, re-execution must impose a _lower_ gas limit than Ethereum, because it must be possible to emulate execution of the transaction (which is more expensive than executing it directly) within a single Ethereum transaction.
```

> **More implementation flexibility:** Interactive proving allows more flexibility in implementation. All that is necessary is the ability to verify a one-step proof on Ethereum. By contrast, rerexecution approaches are tethered to limitations of the EVM.

Reviewer 2: typo & clarifications
```suggestion
**More implementation flexibility:** Interactive proving allows more flexibility in implementation. All that is necessary is the ability to verify a one-step proof on the parent chain, Ethereum. By contrast, re-execution approaches are tethered to limitations of the Ethereum Virtual Machine (EVM).
```
